File: java/16.02.0_Words_Frequency_LCCI.java
/*
执行用时：148 ms, 在所有 Java 提交中击败了56.53% 的用户
内存消耗：87 MB, 在所有 Java 提交中击败了93.67% 的用户
通过测试用例：24 / 24
*/

class WordsFrequency {
    HashMap<String, Integer> map = new HashMap<String, Integer>();
    public WordsFrequency(String[] book) {
        for (String s: book) {
            map.put(s, map.getOrDefault(s, 0) + 1);
        }
    }
    
    public int get(String word) {
        return map.getOrDefault(word, 0);
    }
}

/**
 * Your WordsFrequency object will be instantiated and called as such:
 * WordsFrequency obj = new WordsFrequency(book);
 * int param_1 = obj.get(word);
 */
 

File: cpp/16.02.0_Words_Frequency_LCCI.cpp
/*
执行用时：240 ms, 在所有 C++ 提交中击败了93.42% 的用户
内存消耗：90.7 MB, 在所有 C++ 提交中击败了23.92% 的用户
通过测试用例：24 / 24
*/

class WordsFrequency {
public:
    unordered_map<string, int> umap;
    WordsFrequency(vector<string>& book) {
        for (auto word: book)
            umap[word]++;
    }
    
    int get(string word) {
        // if (!umap.count(word)) return 0;
        return umap[word];
    }
};

/**
 * Your WordsFrequency object will be instantiated and called as such:
 * WordsFrequency* obj = new WordsFrequency(book);
 * int param_1 = obj->get(word);
 */
